gcloud/pubsub/topic.py

```diff
@@ -113,6 +122,8 @@ def publish(self, message, **attrs):
         :rtype: str
         :returns: message ID assigned by the server to the published message
         """
+        if self.add_timestamp_to_messages:
+            attrs['timestamp'] = '%sZ' % _NOW().isoformat()
```

Comment:
But the same user would have also set the value in the dict so this hypothetical user is likely confused. Silently moving forward is not a great way to help their confusion.

Throw `KeyError` if it's already set?